master/internal/rm/agentrm/agent.go

```diff
@@ -639,10 +717,17 @@ func (a *agent) containerStateChanged(ctx *actor.Context, sc aproto.ContainerSta
 	a.agentState.containerStateChanged(sc)
 }
 
-func (a *agent) summarize(ctx *actor.Context) model.AgentSummary {
+func (a *agent) Summarize() model.AgentSummary {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	return a.summarize()
```

Comment:
style question: if you're just returning the result of summarize() why not call that all under Summarize() directly?

Reply:
as in inline this? summarize() is used more than here, sometimes already hold a lock, sometimes not.